### PR TITLE
oneDNN kernels code cleanup

### PR DESCRIPTION
**Pull request by Silv3S**
### PR types
Others

### PR changes
Others

### Describe
We used to have _onednn_reuse.h_ which included multiple methods and utils related to oneDNN kernels. Goal is to split this header file into smaller, dedicated imports. It will improve readability and linking process. Changes done in this PR:
- unify method signatures which differ after migrating kernels to PHI,
- add _matmul_utils.h_ dedicated for `matmul`, `matmul_v2` and `fused_matmul` kernels,
- rename _matmul_v2_mkldnn_op.cc_  to _matmul_mkldnn_op.cc_ as `matmul_v2` kernel was migrated to PHI and this file has only `matmul_v1` kernels left,
- split `SetOutMemDescWithLogicalLayoutFusesSupport` to smaller methods, because only `transpose2` can have all 3 attributes used in this method. Both `fc` and `elementwise` kernels have support just for one attribute, so it can be skipped.

### CODE REVIEW

File: paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc
Comment (tsocha) [inline paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc:159]: It would be great if @jczaja could answer this question :)

File: paddle/phi/backends/onednn/matmul_utils.h
Comment (tsocha) [inline paddle/phi/backends/onednn/matmul_utils.h]: Lets make it all inline.

File: paddle/phi/backends/onednn/matmul_utils.h
Comment (tsocha) [inline paddle/phi/backends/onednn/matmul_utils.h]: I wonder if we need all this code in `.hpp` file.
Could you split it?
Lets keep definitions in `.cpp`.

File: paddle/phi/backends/onednn/matmul_utils.h
Comment (tsocha) [inline paddle/phi/backends/onednn/matmul_utils.h]: @jczaja could you explain this?

File: paddle/phi/kernels/onednn/elementwise_kernel.cc
Comment (tsocha) [inline paddle/phi/kernels/onednn/elementwise_kernel.cc:118]: This switch is shady :D

**Review by tsocha — CHANGES_REQUESTED** [review]

File: paddle/phi/backends/onednn/matmul_utils.h
Comment (Silv3S) [reply paddle/phi/backends/onednn/matmul_utils.h]: Thank you for suggestion, done 👍

**Review by tsocha — DISMISSED** [review]

**Comment by Silv3S** [comment]
@zyfncg could you please review and approve `PR-CI-APPROVAL`? I just changed filename  _matmul_v2_mkldnn_op.cc_ -> _matmul_mkldnn_op.cc_ but CI treats it as newly registered kernel.

**Review by zyfncg — DISMISSED** [review]

**Comment by Silv3S** [comment]
@zyfncg thank you very much for review. Could `PR-CI-Coverage` be manually marked? This PR didn't introduce new code, just moved methods to different directories. All unit tests are passing.

**Comment by zyfncg** [comment]
> @zyfncg thank you very much for review. Could `PR-CI-Coverage` be manually marked? This PR didn't introduce new code, just moved methods to different directories. All unit tests are passing.

@XieYunshen Could you please review this PR for `PR-CI-Coverage`?

**Review by tsocha — DISMISSED** [review]

**Comment by Silv3S** [comment]
I think that problems with low coverage may occur due to our naming unification. Along with PHI migration, we started renaming `MKLDNN` to `oneDNN`. Also new unit tests now start with _test_onednn__ instead of _test_mkldnn__ and they were not included in ctest.

**Review by tsocha — APPROVED** [review]

**Review by zyfncg — DISMISSED** [review]

File: paddle/phi/backends/onednn/matmul_utils.h
Comment (jczaja) [reply paddle/phi/backends/onednn/matmul_utils.h]: It is a question that came up in my head during debugging some longer time ago. I do not have more info than what is in the comment

File: paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc
Comment (jczaja) [reply paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc:159]: it would be great:)

**Review by jczaja — DISMISSED** [review]
LGTM

File: paddle/phi/backends/onednn/matmul_utils.h
Comment (tsocha) [reply paddle/phi/backends/onednn/matmul_utils.h]: The code is not a place for such comments...
Please create a ticket to explain it and remove this TODO from the code.

File: paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc
Comment (tsocha) [reply paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc:159]: It seems that there is no reason why it can't be done for int8, please remove this comment and create a ticket for it.

**Review by tsocha — CHANGES_REQUESTED** [review]
~TODOs have to be removed.~ resolved

**Review by tsocha — DISMISSED** [review]

**Comment by Silv3S** [comment]
Problems with low coverage are resolved. @zyfncg could you approve once more and merge this PR?

**Review by zyfncg — APPROVED** [review]

**Pull request merged**
